sdv/model_generator/main.py

```diff
@@ -30,6 +30,73 @@
 )
 
 
+def generate_model(
+    language: str,
+    input_file_path: str,
+    name: str = "vehicle",
+    include_dir: list[str] = [],
+    target_folder: str = "./gen_model",
+    strict: bool = True,
+    overlays: list[str] = [],
+    extended_attributes: str = "",
+):
+    """
+    Generate model using the passed arguments.
+
+    Args:
+        language (str): The target language of the generated code,
+        e.g. [python, cpp],
+        input_file_path (str): The file to convert. Currently supports JSON
+        and Vspec file formats,
+        include_dir (list[str]): Add include directory to search for included
+        vspec files,
+        target_folder str: The folder name (with relative path) where the code
+        will be generated,
+        strict (bool): Use strict checking or not,
+        overlays (list[str]): Add overlays that will be layered on top of
+        the VSS file in the order they,
+        extended-attributes (str): Whitelisted extended attributes
+        as comma separated list
+    """
+    include_dirs = ["."]
+    include_dirs.extend(include_dir)
+
+    ext_attributes_list = extended_attributes.split(",")
```

Comment:
Extended attributes should already be passed as list; splitting at comma is just argument parsing logic.